### PR TITLE
Install docs dependencies in setup.py

### LINKED ISSUE #3435: Documentation for Pushover reporter says "sends email"
The documentation for the [Pushover reporter](https://docs.buildbot.net/current/manual/cfg-reporters.html#pushover-notifications) talks about sending emails:

> The following simple example will send an email upon the completion of each build. The email contains a description of the Build, its results, and URLs where more information can be obtained.

It should probably talk about sending a Pushover message or something like that.

### PR DESCRIPTION
**Pull request by adityadivekar03**
Remove the pip install command from the docs Makefile used for installing the
sphinx distribution. 
Add the distribution to the extras in setup.py so that it can be installed if required by the user. 

fixes #3435

### CODE REVIEW

**Comment by tardyp** [comment]
I am the culprit for that pip install.

We need to also update the metabuildbot, and maybe other documentation automation (I am not sure if there is an ansible script to build the doc)

so that they use pip install buildbot[docs] (isn't it even possible to run pip install -e . with optional deps)

File: master/setup.py [hunk 1]
```diff
@@ -418,6 +418,9 @@ def define_plugin_entries(groups):
         'bundle': [
             'buildbot-www',
             'buildbot-slave'
+        ],
+        'docs': [
+            'sphinxcontrib-blockdiag>=0.8',
```
Comment (tardyp) [inline master/setup.py]: this is not the same deps.
old version said 

sphinxcontrib-blockdiag (implicit latest)
'docutils>=0.8'  forces upgrade of docutils

**Comment by tardyp** [comment]
travis fails

File: master/docs/Makefile [hunk 2]
```diff
@@ -10,8 +10,6 @@ TAR_TRANSFORM := $(if $(filter bsdtar,$(TAR_VERSION)),-s /^html/$(VERSION)/,--tr
 docs.tgz: clean html singlehtml images-png
 	sed -e 's!href="index.html#!href="#!g' < _build/singlehtml/index.html > _build/html/full.html
 	tar -C _build $(TAR_TRANSFORM) -zcf $@ html
-pipdeps:
```
Comment (rutsky) [inline master/docs/Makefile]: Target `pipdeps` is used in other places in this Makefile.

**Comment by rutsky** [comment]
You need to install Buildbot in `.travis.yml` with `[docs]` extras (in addition to installing `[docs]` in metabuildbot, as @tardyp noted).

Perhaps there is a way to check that Buildbot is installed with `[docs]` extras in `master/docs/Makefile`?

**Comment by adityadivekar03** [comment]
@rutsky  I was thinking that we could use the ```test -e``` command along with the ```shell```  function and run it to check if the user has the ```[docs]``` dependencies installed or not.  We could run the shell command from the makefile, and if the ```[docs]``` dependencies are installed , we could allow the further execution of the makefile for sphinx documentation. This will however involve spawning a new shell, and I would like your opinion whether it can be used. 

About handling the rest of the pipdeps, once we make sure that the user has installed the ```[docs]```, we could remove the pipdeps commands from the rest of the makefile, as it will only be executed if the ```[docs]``` are installed (because of the condition we put in).

@tardyp I somehow skipped the docutils part. I will correct it in the next commit.

**Comment by tardyp** [comment]
@adityadivekar03 we will be very glad if you can finish this PR! Every little helps!

**Comment by adityadivekar03** [comment]
Yes I really do intend to finish it. I was actually waiting for @tardyp 's opinion on the approach I gave in the comments section! Could you give your opinion if possible? :)

**Comment by tardyp** [comment]
I think we should rather modify conf.py to check for python deps, and fail with an explicit message, asking for install of [docs]
I prefer python code to makefile :)

**Comment by adityadivekar03** [comment]
Yes, editing the makefile could be a bit cumbersome and difficult.
Your idea sounds better. I'll try implementing it and definitely revert soon!

**Comment by adityadivekar03** [comment]
@tardyp I've added try-except block to check if the dependencies exist in the conf.py file
<pre><code>
try:
    import sphinxcontrib.blockdiag
    import pkg_resources
    pkg_resources.require('docutils>=0.8')
except DistributionNotFound:
    print("Please install the [docs] dependencies first")
except VersionConflict: 
    print("The version of docutils is not supported. Please install a version >= 0.8")
</code></pre>

Is it okay to just print the above or do you want to catch the error differently?

**Comment by tardyp** [comment]
As [] not well known feature. I'll recommend to print the full command pip
install

Le sam. 5 mars 2016 08:38, adityadivekar03 <notifications@github.com> a
écrit :

**Comment by adityadivekar03** [comment]
@tardyp  There were conflicts in the original MR, so I am putting in a new MR.

File: master/setup.py [hunk 3]
```diff
@@ -426,9 +426,16 @@ def define_plugin_entries(groups):
             'buildbot-www',
             'buildbot-slave'
         ],
+<<<<<<< HEAD
```
Comment (tardyp) [inline master/setup.py]: you forgot to fix the merge conflict issue.

**Comment by tardyp** [comment]
you can just amend your commit, and forcepush the result on the same branch. its better like this.

File: .travis.yml [hunk 4]
```diff
@@ -128,7 +128,11 @@ install:
   # astroid > 1.4 is incompatible with pylint 1.1.0
   - "[ $TESTS != lint ] || pip install pylint==1.1.0 pep8==1.5.7 flake8  astroid==1.3.8"
   # TODO: Pin Sphinx version to workaround http://trac.buildbot.net/ticket/3408
-  - "[ $TESTS != docs ] || pip install sphinx==1.3.3"
+  - |
+      [ $TESTS != docs ] || \
+      pip install sphinx==1.3.3 \
```
Comment (tardyp) [inline .travis.yml]: why not doing pip install -e ./master[docs] ?

Comment (tardyp) [inline .travis.yml]: that would require also to add sphinx to [docs] and add more coherency

**Comment by adityadivekar03** [comment]
I apologize for all the patchy work. And I have opened a new MR for this. I tried rebasing before, and applying my patch on top of it. It did not occur to me that I could have added the lines in setup.py myself.
And I've made the changes in travis.yml as per your recommendation, and added the sphinx package to ```[docs]```

**Pull request closed without merging**